lib/stdio/encoding.js

```diff
@@ -5,21 +5,22 @@ import {isUint8Array} from './utils.js';
 // Apply the `encoding` option using an implicit generator.
 // This encodes the final output of `stdout`/`stderr`.
 export const handleStreamsEncoding = (stdioStreams, {encoding}, isSync) => {
-	if (stdioStreams[0].direction === 'input' || IGNORED_ENCODINGS.has(encoding) || isSync) {
-		return stdioStreams.map(stdioStream => ({...stdioStream, encoding}));
-	}
-
-	const transform = encodingEndGenerator.bind(undefined, encoding);
-	const objectMode = stdioStreams.findLast(({type}) => type === 'generator')?.value.readableObjectMode === true;
```

Comment:
The following line basically indicates: uses the same `objectMode` as the previous generator, or `false` if none. This PR simplifies this by using an internal `"previous"` value, which is handled below.